forest/src/main.rs

```diff
@@ -36,11 +39,32 @@ async fn main() {
         }
     };
 
-    let lp2p_service = Libp2pService::new(logger, &config.network, net_keypair);
+    let running = Arc::new(AtomicUsize::new(0));
+    let r = running.clone();
+    ctrlc::set_handler(move || {
+        let prev = r.fetch_add(1, Ordering::SeqCst);
+        if prev == 0 {
+            println!("Got interrupt, shutting down...");
+        } else {
+            process::exit(0);
+        }
+    })
+    .expect("Error setting Ctrl-C handler");
 
-    task::block_on(async move {
+    // Start libp2p service
+    let lp2p_service = Libp2pService::new(logger, &config.network, net_keypair);
```

Comment:
I would just call it p2p_service

Reply:
unchanged in this PR, but changed it haha